tensorzero-core/src/db/clickhouse/dataset_queries.rs

```diff
@@ -655,6 +574,246 @@ impl DatasetQueries for ClickHouseConnectionInfo {
 
         Ok(datapoints)
     }
+
+    /// Inserts a batch of datapoints into the database. Internally, separate chat and JSON datapoints and write them to the appropriate tables. Note that this is not very atomic: the Chat table and Json table updates are not rolled back if one fails.
+    ///
+    /// Requires us to take ownership of each datapoint in `datapoints` since we split them into two Vecs internally.
```

Comment:
I don't think you're actually taking ownership here since you're pushing pointers but I might be wrong.

Reply:
Yep, you're right, this was from earlier. Fixed.